src/services/MessagePreviewService.ts

```diff
@@ -0,0 +1,46 @@
+import { Message, TextChannel, MessageEmbed } from "discord.js";
+
+class MessagePreviewService {
+	private static instance: MessagePreviewService;
+
+	/* eslint-disable */
+	private constructor() {}
+	/* eslint-enable */
+
+	static getInstance(): MessagePreviewService {
+		if (!this.instance) {
+			this.instance = new MessagePreviewService();
+		}
+
+		return this.instance;
+	}
+
+	async generatePreview(link: string, callingMessage: Message): Promise<void> {
+		const msgArray = this.stripLink(link);
+
+		if (this.verifyGuild(callingMessage, msgArray[0])) {
+			if (callingMessage.guild?.available) {
+				const channel = callingMessage.guild.channels.cache.get(msgArray[1]) as TextChannel;
+				const messageToPreview = await channel.messages.fetch(msgArray[2]);
+
+				const embed = new MessageEmbed()
+					.setAuthor(messageToPreview.member?.nickname || messageToPreview.author.username, messageToPreview.author.avatarURL() || undefined, link)
+					.addField(`Called by ${callingMessage.member?.nickname || callingMessage.author.username}`, `[Click for context](${link})`)
+					.setDescription(`${messageToPreview.content}\n`)
+					.setColor(messageToPreview.member?.displayColor || "#FFFFFE");
```

Comment:
It bugs me that this uses chaining rather than doing `embed.method()` like all the other embeds. @Jason2605 are you okay with this or would you like it changed for consistency?

Reply:
chaining is prettier, fix the others